prow/sidecar/run.go

```diff
@@ -125,7 +129,12 @@ func (o Options) Run(ctx context.Context) (int, error) {
 				// second upload but we can tolerate this as we'd rather get SOME
 				// data into GCS than attempt to cancel these uploads and get none.
 				logrus.Errorf("Received an interrupt: %s, cancelling...", s)
-				cancel()
+
+				err = o.doUpload(spec, false, true, metadata, buildLogs)
```

Comment:
This case skips some of the steps before `doUpload` is called, in particular checking for the deprecated wrapper options and censoring the data before upload. Please put all this common logic in a function that can be called in both places.

Reply:
Added a new method `preUpload` which performs steps required before upload and call it before both upload process.